prow/config/config.go

```diff
@@ -864,21 +882,33 @@ func mergeJobConfigs(a, b JobConfig) (JobConfig, error) {
 	return c, nil
 }
 
-func setPresubmitDecorationDefaults(c *Config, ps *Presubmit) {
+func setPresubmitDecorationDefaults(c *Config, ps *Presubmit, repo string) {
 	if ps.Decorate {
-		ps.DecorationConfig = ps.DecorationConfig.ApplyDefault(c.Plank.DefaultDecorationConfig)
+		dcByRepo := c.Plank.GetDefaultDecorationConfigs(repo)
+		def := dcByRepo.ApplyDefault(c.Plank.DefaultDecorationConfig)
+		ps.DecorationConfig = ps.DecorationConfig.ApplyDefault(def)
 	}
 }
 
-func setPostsubmitDecorationDefaults(c *Config, ps *Postsubmit) {
+func setPostsubmitDecorationDefaults(c *Config, ps *Postsubmit, repo string) {
 	if ps.Decorate {
-		ps.DecorationConfig = ps.DecorationConfig.ApplyDefault(c.Plank.DefaultDecorationConfig)
+		dcByRepo := c.Plank.GetDefaultDecorationConfigs(repo)
+		def := dcByRepo.ApplyDefault(c.Plank.DefaultDecorationConfig)
+		ps.DecorationConfig = ps.DecorationConfig.ApplyDefault(def)
 	}
 }
 
 func setPeriodicDecorationDefaults(c *Config, ps *Periodic) {
 	if ps.Decorate {
-		ps.DecorationConfig = ps.DecorationConfig.ApplyDefault(c.Plank.DefaultDecorationConfig)
+		var org, repo string
+		if len(ps.UtilityConfig.ExtraRefs) > 0 {
+			org = ps.UtilityConfig.ExtraRefs[0].Org
+			repo = ps.UtilityConfig.ExtraRefs[0].Repo
+		}
+
+		dcByRepo := c.Plank.GetDefaultDecorationConfigs(fmt.Sprintf("%s/%s", org, repo))
```

Comment:
I think the case when we **don't** have ExtraRefs should be handled more elegantly than passing a `"/"` string to `GetDefaultDecorationConfigs`, hoping that it will do the right thing. Either pass a `"*"` there, or maybe even not call `GetDefaultDecorationConfigs` and use `c.Plank.DefaultDecorationConfigs["*"]` directly.

Reply:
addressed